src/pydicom/pixels/decoders/native.py

```diff
@@ -132,19 +137,49 @@ def _rle_decode_frame(
     nr_segments = len(offsets)
 
     # Check that the actual number of segments is as expected
-    bytes_per_sample = nr_bits // 8
-    if nr_segments != nr_samples * bytes_per_sample:
+    if nr_bits == 1:
+        if nr_samples != 1:
+            raise ValueError(
```

Comment:
Maybe `NotImplementedError` instead

Reply:
I have made this change in [c1e9a0f](https://github.com/pydicom/pydicom/pull/2216/commits/c1e9a0f2c1b8225b608765a11be870fdd5d7fc99) to make it in line with other similar errors. Though I'm not sure it's the best in my opinion, for two reasons:

- My understanding of `NotImplementedError` is that it is conventionally used in narrower situations: where a subclass doesn't implement a method implied by a superclass. See [docs](https://docs.python.org/3/library/exceptions.html#NotImplementedError). But I suppose that a library can decide how to use it
- It implies that this could be implemented but hasn't been. But in reality this situation is not defined in the standard, so it's not clear what implementing it would even mean. Really it's just non-conformant data